tests/callbacks/test_gpu_stats_monitor.py

```diff
@@ -63,6 +63,37 @@ def test_gpu_stats_monitor(tmpdir):
         assert any(f in h for h in met_data.dtype.names)
 
 
+@RunIf(min_gpus=1)
+def test_gpu_stats_monitor_no_queries(tmpdir):
+    """
+    Test GPU stats are logged using a logger.
+    """
+    model = BoringModel()
+    gpu_stats = GPUStatsMonitor(
+        memory_utilization=False,
+        gpu_utilization=False,
+        intra_step_time=True,
+        inter_step_time=True,
+        fan_speed=False,
+        temperature=False,
+    )
+    logger = CSVLogger(tmpdir)
+    log_every_n_steps = 2
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=2,
+        limit_train_batches=7,
+        log_every_n_steps=log_every_n_steps,
+        gpus=1,
+        callbacks=[gpu_stats],
+        logger=logger,
+    )
+
+    trainer.fit(model)
+    assert trainer.state.finished, f"Training failed with {trainer.state}"
```

Comment:
Can you add assertions for the `CSVLogger` state?

Reply:
same as the test just above (i.e. test_gpu_stats_monitor? )

Reply:
minus the `fields = ["utilization.gpu", "memory.used", "memory.free", "utilization.memory"]`

Reply:
i guess see 0579717755a97ece28abf10dd21341683ae4b03c if it's ok?

Reply:
Actually, I would avoid using the `CSVLogger` entirely and `unittest.mock.patch` the `trainer.logger.log_metrics` method which is called by the `GPUStatsMonitor` 

https://github.com/PyTorchLightning/pytorch-lightning/blob/ad3f183bc367095b3f121882489fb532983c672a/pytorch_lightning/callbacks/gpu_stats_monitor.py#L179

Are you familiar with mocking and asserting its calls? Ping if you need any help

Reply:
not exactly, but i can get up to speed

Reply:
is this what you had in mind?

Reply:
Pushed https://github.com/PyTorchLightning/pytorch-lightning/pull/9014/commits/fe7416ccb8e918ee4fbe48e1debfeb74c4c1db2b with what I had in mind 👍